index.d.ts

```diff
@@ -131,7 +136,20 @@ interface TwitterOptions {
   /** access token secret from your User (oauth_token_secret) */
   access_token_secret?: OauthTokenSecret;
   /** bearer token */
-  bearer_token?: string;
+  bearer_token?: never;
+}
+
+interface TwitterOptionsAppAuthentication extends TwitterOptionsBasics {
+  /** consumer key from Twitter. */
+  consumer_key?: never;
+  /** consumer secret from Twitter */
+  consumer_secret?: never;
+  /** access token key from your User (oauth_token) */
+  access_token_key?: never;
+  /** access token secret from your User (oauth_token_secret) */
+  access_token_secret?: never;
```

Comment:
defining all the keys as "never" seems a bit useless to me. If we were to define the `TwitterOptionsAppAuthentication` and `TwitterOptionsUserAuthentication` as types and not interfaces, we could just leave out anything that's not supposed to be included.

Reply:
Thank's @T0TProduction for your feedback.

The problem is that unions in TypeScript are not exclusive (see https://github.com/Microsoft/TypeScript/issues/14094).
This means that if we leave out the nevers, an object with a `bearer_token` and a `consumer_key` would be a valid `TwitterOptions` object.

This StackOverflow comment sums the problem up: https://stackoverflow.com/a/61281828

I tried to use types instead of interfaces as you described it, but the problem remained the same.

In my opinion, we should use `never` to have stricter rules.

Reply:
Ah, I see!
I agree, if we do not want to allow both at the same time, that's the best (and only) option.

I just checked [the code](https://github.com/draftbit/twitter-lite/blob/084a68d23be3d4485d814abc424b42dd00b163be/twitter.js#L66), and now I'm a bit confused though: 
according to my understanding of the constructor, it *could* actually handle both options ( but set the type to 'App' none the less )
Maybe it should ignore keys after already having a token; or throw warnings/errors when multiple credentials are added simultaneously. 
Then the types/interfaces should be exactly as you said, to support that choice of exclusivity